### DISCUSSION TITLE
Pastas WellModel function - problems with more than three wells

### DISCUSSION THREAD

**Discussion by pastapepijn** [pastas/pastas#614] [Q&A] [answered]
Hi,

I have used the function ps.WellModel to include multiple wells in my pastas model. In total, I want to add five wells with a constant extraction every day. When I add two or three wells (in m^3/day *10^-6 as I saw in the ReadTheDocs example, is this correct?) it all seems to function properly but as soon as the fourth well is included I get some unreliable data. Below are three added wells. 
![image](https://github.com/pastas/pastas/assets/141640440/ae5d0af3-1fd5-443d-9f17-6e79c93341da)
![image](https://github.com/pastas/pastas/assets/141640440/0b032850-032c-4897-90b7-c0c9f6ff1b27)
Besides the explainable noise, this all looks alright. But when I add well 4 which has a relatively low extraction and big distance (see figure below) the whole fit totally changes where precipitation patterns are not even visible, even though the changes should be minor because of the distance and low extraction.
![image](https://github.com/pastas/pastas/assets/141640440/272d907d-5fc3-4239-a2e7-314561dc4b10)
![image](https://github.com/pastas/pastas/assets/141640440/4be0754a-5728-41e6-b5fc-b9db62726feb)
When well five is added with significantly more daily extraction and approximately the same distance it does give something reliable. Although, the fit is getting worse than with three wells. 
![image](https://github.com/pastas/pastas/assets/141640440/36931ef5-67cd-4121-b9cf-d9ab5da45f17)
![image](https://github.com/pastas/pastas/assets/141640440/b30b22c4-2c0a-45be-aaf2-828e10cdefbc)
What is going wrong here? Why does well 4 not work. (below the csv with extraction data)
![image](https://github.com/pastas/pastas/assets/141640440/f4669ca2-8ad5-41d3-a7b2-e9d1b67bebd2)

**Comment by mbakker7** [comment]
Do I understand correctly that the discharge of none of the wells varies through time?

**Reply by pastapepijn** [reply]
Yes that is correct. It is an estimate of daily domestic consumptions in Nepal. Accurate daily data is not available.

**Comment by dbrakenhoff** [comment, marked as answer]
Hi @pastapepijn,

Thanks for your question(s)!

> in m^3/day *10^-6 as I saw in the ReadTheDocs example, is this correct?

The parameter `A` can sometimes become quite small when the units of the well pumping time series are m3/day and the well discharge is large. The gain (lowering of the head per unit of pumping) is then m / (m3/day). For very large extractions it can be more useful to convert the units of the input time series to e.g. Mm3/yr. That way the unit of the gain becomes m / (Mm3/day) which can be easier interpret. But as long as you're consistent in what you put in and how you interpret the output, any choice is fine. In [this example](https://pastas.readthedocs.io/en/latest/examples/multiple_wells.html) I convert the units of the gain from m / (m3/d) (this is the unit of the original pumping time series) to m / (Mm3/yr) because that makes for a nicer unit to present the results in. 

As for the question why 4 wells is causing your result to change, this is difficult to say based on these results, but I have some remarks:

- First of all, your head time series is relatively short, and your well discharge rates are constant in each well. This means there is relatively little information in the head and pumping time series for the model to find a strong relationship between the two. As your 3-well model shows the impact of the pumping wells is increasing towards the end of your time series. If you extend your simulation, you might see that this estimate of the impact of the wells is way too large(?)... It might be the case that the pumping well time series in your model are used to fit the downwards trend in the observation data, but there isn't much certainty about that trend. Maybe interpolation is possible with this model, but extrapolation will be very uncertain. 

- I agree that fitting it with 4 wells shouldn't lead to a worse model than with 3 wells. The fact that it performs worse might be related to starting parameters though, especially since the fit seems quite uncertain (see previous point). A couple things you can try:
   - Use the optimized parameters from the 3-well model as starting parameters for the 4-well model, e.g. `ml.set_parameter("well_A", initial=A_3wellmodel)`. Then solve with `ml.solve(initial=False)` to make pastas use your estimates of the initial parameters.
   - Try solving without a noise-model: `ml.solve(noise=False)`
   - Try using the two-step solving method to obtain better starting parameters. See [this example for how this is done](https://pastas.readthedocs.io/en/latest/examples/calibration_options.html#Improving-initial-parameters).

Hope this helps a bit, and let us know if you have any further questions!

**Reply by dbrakenhoff** [reply]
@pastapepijn, can we mark this discussion as answered?

**Reply by pastapepijn** [reply]
yes
